app/state/reducers/ui/reservationInfoModalReducer.js

```diff
@@ -0,0 +1,36 @@
+import Immutable from 'seamless-immutable';
+
+import types from 'constants/ActionTypes';
+
+const initialState = Immutable({
+  reservation: null,
+  show: false,
+});
+
+function reservationInfoModalReducer(state = initialState, action) {
+  switch (action.type) {
+
+    case types.API.RESERVATION_PUT_SUCCESS: {
+      return state.merge({
+        reservation: action.payload,
+      });
+    }
+
+    case types.UI.SHOW_RESERVATION_INFO_MODAL: {
+      return state.merge({
+        reservation: action.payload,
+        show: true,
+      });
+    }
+
+    case types.UI.HIDE_RESERVATION_INFO_MODAL: {
+      return initialState;
```

Comment:
- [ ] Currently when the modal is closed the modal becomes empty while it is disappearing. This could be fixed by not setting `reservation` to null here, i.e. changing this to `return state.merge({ show: false });`.